app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/DatasetsManagementInterface.kt

```diff
@@ -0,0 +1,31 @@
+package com.github.HumanLearning2021.HumanLearningApp.model
+
+import com.github.HumanLearning2021.HumanLearningApp.firestore.FirestoreDatasetInterface
+
+interface DatasetsManagementInterface {
+
+    /**
+     * Retrieves the dataset
+     *
+     * @param key - the unique key of the desired dataset
+     * @return an instance of FirestoreDatasetInterface
+     * @throws IllegalArgumentException: if there exists no dataset of the specified name
+     */
+    fun getDataset(key: String): FirestoreDatasetInterface
+
+    /**
+     * Initializes a new dataset
+     *
+     * @param key - the unique key of the desired dataset
+     * @param categories - the set of categories the dataset contains, can be empty
+     */
+    fun initializeDataset(key: String, categories: Set<Category>): FirestoreDatasetInterface
```

Comment:
1. clients should not generate unique keys themselves
2. What happened to human-readable name?

Reply:
True, mistake on my part. I will put it back as human-readable dataset name